### DISCUSSION TITLE
Limit objective amount config and repRequired functionality for Objectives

### DISCUSSION THREAD

**Discussion by Sazuzaki** [ejektaflex/Bountiful#265] [Mod Ideas]
I was tweaking some rewards, and wanted to up the emerald amount. I did it to around a maximum of 32, and ended up getting bounties with like 10+ objectives. I know it has something to do with the calculations between obj worth and reward worth, but I wanted to make the harder objectives reward more emeralds without removing them from the early game. Ex: A bounty to slay 4 zombies rewards 2 emeralds, but a bounty to slay a dragon rewards 32. I thought being able to limit the amount of objectives that can appear in a single bounty could 
alleviate this problem.

I also wanted to ask if you could make the repRequired field work for objectives too. It would be nice to have objectives get progressively harder and rewards get better as you rank up your reputation.

Thanks for the hard work, and hope to see more.

**Comment by ejektaflex** [comment]
Hi there! It seems that you'll need to add some more high value objectives, that's why you're getting so many.

Details on how bounties are generated can be found here:
https://kambrik.ejekta.io/mods/bountiful/advanced/generation

TL;DR:
Rewards are generated first, and then objectives are generated to match. However, if the generated objectives STILL do not match the value of the rewards, the generator will aggressively add new objectives until the value is reasonably close.

This is because Bountiful wants to make balanced rewards as high of a priority as possible - as such, I presume that you have some high value rewards and some very low value objectives, so it needs to create a LOT of objectives to match the rewards. I'd suggest adding some more higher value objectives.

Because of our focus on balance-first, there is no way to guarantee a limited number of objectives. However, the game tries to limit the number to 1-2 objectives by default. If you are seeing more than three, then you need to add balanced objectives to match your rewards, as mentioned above.

As for repRequired, that's something that I'll look into!

**Reply by Sazuzaki** [reply]
Gotcha, thanks for clarifying that. I guess I skimmed that on the wiki and it didn't stick, but I'd probably also have to make the higher value objectives more common too in order prevent the scenario I mentioned. 

Thanks for the quick and informative reply! Appreciate you.
